jumanji/environments/routing/sokoban/env.py

```diff
@@ -0,0 +1,593 @@
+                mask_target_box,
+                TARGET_BOX,
+                jnp.maximum(variable_grid, fixed_grid),
+            ),
+        ).astype(jnp.uint8)
+
+        return single_grid
+
+    def level_complete(self, state: State) -> chex.Array:
+        """
+        Checks if the sokoban level is complete.
+
+        Args:
+            state: `State` object representing the current state of the environment.
+
+        Returns:
+            complete: Boolean indicating whether the level is complete
+            or not.
+        """
+        return self.reward_fn.count_targets(state) == N_BOXES
+
+    def check_space(
+        self,
+        grid: chex.Array,
+        location: chex.Array,
+        value: int,
+    ) -> chex.Array:
+        """
+        Checks if a specific location in the grid contains a given value.
+
+        Args:
+            grid: Array (uint8) shape (num_rows, num_cols) The grid to check.
+            location: Tuple size 2 of Array (int32) shape () containing the x
+            and y coodinate of the location to check in the grid.
+            value: int The value to look for.
+
+        Returns:
+            present: Array (bool) shape () indicating whether the location
+            in the grid contains the given value or not.
+        """
+
+        return grid[tuple(location)] == value
+
+    def in_grid(self, coordinates: chex.Array) -> chex.Array:
+        """
+        Checks if given coordinates are within the grid size.
+
+        Args:
+            coordinates: Array (uint8) shape (num_rows, num_cols) The
+            coordinates to check.
+        Returns:
+            in_grid: Array (bool) shape () Boolean indicating whether the
+            coordinates are within the grid.
+        """
+        return jnp.all((0 <= coordinates) & (coordinates < GRID_SIZE))
+
+    def detect_noop_action(
+        self,
+        variable_grid: chex.Array,
+        fixed_grid: chex.Array,
+        action: chex.Array,
+        agent_location: chex.Array,
+    ) -> chex.Array:
+        """
+        Masks actions to -1 that have no effect on the variable grid.
+        Determines if there is space in the destination square or if
+        there is a box in the destination square, it determines if the box
+        destination square is valid.
+
+        Args:
+            variable_grid: Array (uint8) shape (num_rows, num_cols).
+            fixed_grid Array (uint8) shape (num_rows, num_cols) .
+            action: Array (int32) shape () The action to check.
+
+        Returns:
+            updated_action: Array (int32) shape () The updated action after
+            detecting noop action.
+        """
+
+        new_location = agent_location + MOVES[action].squeeze()
+
+        valid_destination = self.check_space(
+            fixed_grid, new_location, WALL
+        ) | ~self.in_grid(new_location)
+
+        updated_action = jax.lax.select(
+            valid_destination,
+            -jnp.ones(shape=(), dtype=jnp.int32),
```

Comment:
I think we should use the constant here, not 100% this will return the correct type, but seems like default int, so should be fine

```suggestion
            NOOP,
```